src/convert/transformers/decomposedMetadataTransformer.ts

```diff
@@ -47,6 +47,12 @@ export class DecomposedMetadataTransformer extends BaseMetadataTransformer {
         component,
         children: new ComponentSet([], this.registry),
       };
+      if (component.xml && existing.component && !existing.component.xml) {
+        // we've already found and created the parent of this component on L~38
+        // but now we have more information about the parent (xml) that we didn't have before, so add it
+        // TODO: find way to determine which component has more information
+        existing.component = component;
+      }
```

Comment:
Is the TODO doable now or is it big enough to warrant its own WI and a separate effort?

I like the comment!